src/main/java/org/mskcc/cbio/portal/dao/DaoSampleProfile.java

```diff
@@ -50,98 +60,62 @@ public final class DaoSampleProfile {
     private DaoSampleProfile() {}
 
     private static final int NO_SUCH_PROFILE_ID = -1;
-    private static final String TABLE_NAME = "sample_profile";
-
-    public static int addSampleProfile(Integer sampleId, Integer geneticProfileId, Integer panelId) throws DaoException {        
-        if (MySQLbulkLoader.isBulkLoad()) {
-
-            // Add new record using bulk loader. Order of fields is:
-            // 1. sample ID
-            // 2. genetic Profile ID
-            // 3. gene panel ID
-            if (panelId != null) {
-                MySQLbulkLoader.getMySQLbulkLoader(TABLE_NAME).insertRecord(
-                    Integer.toString(sampleId),
-                    Integer.toString(geneticProfileId),
-                    Integer.toString(panelId));
-            } else {
-                MySQLbulkLoader.getMySQLbulkLoader(TABLE_NAME).insertRecord(
-                    Integer.toString(sampleId),
-                    Integer.toString(geneticProfileId),
-                    null);
-            }
 
-            return 1;
-        }
+    public static void upsertSampleToProfileMapping(Collection<Integer> sampleIds, Integer geneticProfileId, Integer panelId) throws DaoException {
```

Comment:
There are not too many records involved here (# of samples) * (# of profiles), but I do wonder whether switching from the bulk loader to individual SQL statements will have a performance impact. I guess we will see.